More SUMA 5.0 containerization fixes (#2895)
uyuni-project/uyuni-docs: modules/installation-and-upgrade/pages/container-deployment/suma/sle-micro-deployment.adoc

endif::[]

In this section, you will gain expertise in setting up and deploying a containerized {productname} {productnumber} Server.

> This process encompasses the installation of {micro}, and then initiating interaction with the container through the
> container utilities [command]``mgradm`` and [command]``mgrctl``.

Reviewer: ```suggestion
This process installs {micro}, and then interacts with the container through the
container utilities [command]``mgradm`` and [command]``mgrctl``.
```

== Container Host general Requirements

> For general requirements see, xref:installation-and-upgrade:general-requirements.adoc[General requirements].

Reviewer: ```suggestion
For general requirements, see xref:installation-and-upgrade:general-requirements.adoc[].
```

A {micro} or {sles} server should be installed from installation media.
This procedure is described below.

[[installation-server-containers-requirements]]
== Container Host Requirements

> For CPU, RAM and storage requirements see, xref:installation-and-upgrade:hardware-requirements.adoc[Hardware requirements].

Reviewer: ```suggestion
For CPU, RAM, and storage requirements, see xref:installation-and-upgrade:hardware-requirements.adoc[].
```

== {micro} {microversion} Installation

> Get a copy of the {micro} {microversion} installation media. You can begin the installation on either a virtual machine or on a physical server.

Reviewer: ```suggestion
Get a copy of the {micro} {microversion} installation media.
You can begin the installation on either a virtual machine or on a physical server.
```

.Procedure: {micro} {microversion} installation
. Begin the installation from the installation media.

> . Adjust Keyboard and language. Click the [systemitem]``checkbox`` to accept the License Agreement.

Reviewer: ```suggestion
. Adjust keyboard and language.
  Click the [systemitem]``checkbox`` to accept the license agreement.
```

> . Click [systemitem]``Next`` to continue.

Reviewer: ```suggestion
. Click btn:[Next] to continue.
```

> . Select your registration method. For this example we will register the sever with the {scc}.

Reviewer: ```suggestion
. Select your registration method.
  For this example, we will register the sever with the {scc}.
```

> . Enter your {scc} email address.

Reviewer: ```suggestion
. Enter your {scc} E-mail Address.
```

Author: From what I remember, in English, `email` is not written with a dash between the `e` and `mail` like we do it in German. Do you know more here, @jcayouette?

Reviewer 2: It should be: "Enter your {SCC} email address."

Author: Thank you, Joseph. Then I leave my changes.

. Enter your registration code for {micro} {microversion}.
+

.Base operating system for {productname} {productnumber} containerized Server
[NOTE]
====
For SUSE Manager 5.0, the base operating system of the container is {sles} {bci-suma}.
You can find these keys in your {scc} account.
====

. Decide if you would like to activate online repositories or not.

> . Click [systemitem]``Next`` to continue.

Reviewer: ```suggestion
. Click btn:[Next] to continue.
```

. Click [systemitem]``Next`` to continue.

This concludes the installation of {micro} {microversion}.

> == Update the container host

Reviewer: ```suggestion
== Update the Container Host
```

> Once installation completes, update the container host system.

Reviewer: ```suggestion
When the installation completes, update the container host system.
```

.Procedure: Update the container host and install the container utilities

> . Login at the prompt as *root*.

Reviewer: ```suggestion
. Log in at the prompt as *root*.
```

. Run **transactional-update**:
+

[source, shell]
----
transactional-update
----

. Reboot the system.

> . Login as root.

Reviewer: ```suggestion
. Log in as root.
```

. Install the container utilities:
+

[source, shell]
----
transactional-update pkg install mgradm mgrctl mgradm-bash-completion mgrctl-bash-completion
----
+

. Reboot the system.

== Deploy {productname} {productnumber} as container on {micro} {microversion}

=== Basic and advanced deployment with mgradm

.Procedure: Basic deployment of a {productname} {productnumber} container with Podman

> . From the terminal of the container host, run the following command as the root user. Entering your servers FQDN is optional.
> Leave blank for a default setup.

Reviewer: ```suggestion
. From the terminal of the container host, run the following command as the root user.
  Entering your server's FQDN is optional.
  Leave blank for a default setup.
```

====
+

. Enter a database and certificate password when prompted. Press btn:[Enter].

. Enter a certificate and administrator account password when prompted.
+

[NOTE]
====
The administrator account password must be at least 5 characters and less
than 48 characters in length.
====
+

. Press btn:[Enter].

. Wait for deployment to complete.

. Open a browser and proceed to your servers FQDN, or IP address.

In this section you learned the basic method for deploying a {productname} {productnumber} Server container.

.Procedure: Advanced deployment of a {productname} {productnumber} container using a custom configuration file

> . Prepare a configuration file named [filename]``mgradm.yaml``mgradm.yaml`` similar to the following example:

Reviewer: ```suggestion
. Prepare a configuration file named [filename]``mgradm.yaml`` similar to the following example:
```

====
+

> . From the terminal run the following command as the root user. Entering your servers FQDN is optional.

Reviewer: ```suggestion
. From the terminal, run the following command as the root user.
  Entering your server's FQDN is optional.
```

. Wait for deployment to complete.

> . Open a browser and proceed to your servers FQDN, or IP address.

Reviewer: ```suggestion
. Open a browser and proceed to your server's FQDN or IP address.
```

In this section you learned how to deploy a {productname} Server container.

=== Persistent volumes

[NOTE]
====
If you are just testing out {productname} you do not need to specify these volumes. [command]``mgradm`` will setup the correct volumes by default.

Specifying volume locations will generally be used for larger production deployments.
====

Many users will want to specify locations for their persistent volumes.

By default, [command]``Podman`` stores its volumes in [path]``/var/lib/containers/storage/volumes/``. 

You can provide custom storage for the volumes by mounting disks on this path or the expected volume path inside it such as: [path]``/var/lib/containers/storage/volumes/var-spacewalk``.
This is especially important for the database and package mirrors. 

For a list of all persistent volumes in the container see, xref:installation-and-upgrade:container-management/persistent-container-volumes.adoc[Persistent container volumes].

> == Container management

Reviewer: ```suggestion
== Container Management
```

. Wait for deployment to complete.

. Open a browser and proceed to your server's FQDN or IP address.

> In this section you learned how to deploy a {productname} Server container.

Reviewer 3: ```suggestion
In this section you learned how to deploy the {productname} Server container.
```

if {productname} is SUSE Manager, is it correct to  write `a` {productname}, but if this is Uyuni, it should  be `an` {productname}. What about writing `the` {productname} instead?
